source/features/add-co-authored-by.tsx

```diff
@@ -4,7 +4,33 @@ import * as api from '../libs/api';
 import features from '../libs/features';
 import {getOwnerAndRepo, getDiscussionNumber, getOP} from '../libs/utils';
 
-let coAuthors;
+// TODO: See if these types are used elsewhere
+interface Author {
+	email: string;
+	user: {
+		name: string;
+		login: string;
+	};
+}
+
+interface CommitNode {
+	commit: {
+		author: Author;
+	}
+}
+
+interface UserInfo {
+	repository: {
+		pullRequest: {
+			commits: {
+				nodes: CommitNode[];
+			};
+		};
+	}
```

Comment:
RIP ease of contribution 😢

Reply:
I would argue that it actually makes contributions simpler as you can be more confident about your code, you get auto-complete when developing, and you can more easily respond to PR reviews with the powerful rename/refactoring tools available for TS. Verbosity doesn't mean harder, it's just means more writing.

Reply:
The amount of developers familiar with TS is much smaller than JS's. Even if I know everything about RGH, I still have to do a lot of googling to figure out how to type this without making it awful, for example:

```js
select('.a').addEventListener(({target: link}) => link.remove())
```